src/inspector_agent.cc

```diff
@@ -208,27 +213,36 @@ class AgentImpl {
   inspector_socket_t* client_socket_;
   blink::V8Inspector* inspector_;
   v8::Platform* platform_;
-  std::vector<std::string> message_queue_;
-  std::vector<std::string> outgoing_message_queue_;
+  MessageQueue incoming_message_queue_;
+  MessageQueue outgoing_message_queue_;
   bool dispatching_messages_;
+  int frontend_session_id_;
+  int backend_session_id_;
 
   friend class ChannelImpl;
   friend class DispatchOnInspectorBackendTask;
   friend class SetConnectedTask;
   friend class V8NodeInspector;
   friend void InterruptCallback(v8::Isolate*, void* agent);
+  friend void DataCallback(uv_stream_t* stream, ssize_t read,
+                           const uv_buf_t* buf);
 };
 
 void InterruptCallback(v8::Isolate*, void* agent) {
-  static_cast<AgentImpl*>(agent)->PostMessages();
+  static_cast<AgentImpl*>(agent)->DispatchMessages();
+}
+
+void DataCallback(uv_stream_t* stream, ssize_t read, const uv_buf_t* buf) {
+  inspector_socket_t* socket = static_cast<inspector_socket_t*>(stream->data);
+  static_cast<AgentImpl*>(socket->data)->OnRemoteDataIO(socket, read, buf);
```

Comment:
Just trying to follow the libuv conventions. Technically, inspector_socket.{h,cc} is a partial implementation of the WebSockets protocol - so it can either be removed (if a better option is introduced into Node.js) or extended if other clients within Node.js need the protocol.